### PR TITLE
Added backup template and values

### PR DESCRIPTION
**Pull request by ayeshasiddiqa161**

### CODE REVIEW

File: application/values-test.yaml [hunk 1]
```diff
@@ -1294,3 +1294,26 @@ cronJob:
...
+  snapshotVolumes: true
+  storageLocation: "dpa-1"
+  ttl: "1h0m0s"
+  includedResources: 
+    - deployments
+    - services
+    - persistentvolumeclaims
+    - secrets
+  excludedResources:
+    - kube-system
```
Comment (rasheedamir) [inline application/values-test.yaml:1312]: @ayeshasiddiqa161 is this a resource? isn't a namespace name?

File: application/templates/backup.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{{- if (.Values.backup).enabled }}
+apiVersion: velero.io/v1
+kind: Backup
+metadata:
+  name: {{ .Values.backup.backupCRName }}
+  namespace: openshift-adp
+spec:
+  includedNamespaces:
+    - {{ .Values.backup.includedNamespaces | quote }}
```
Comment (rasheedamir) [inline application/templates/backup.yaml:9]: @ayeshasiddiqa161 shouldn't this be fixed to current namespace? why user has to input it?

File: application/templates/backup.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{{- if (.Values.backup).enabled }}
+apiVersion: velero.io/v1
+kind: Backup
+metadata:
+  name: {{ .Values.backup.backupCRName }}
```
Comment (rasheedamir) [inline application/templates/backup.yaml:5]: @ayeshasiddiqa161 can't this be dynamic?

File: application/templates/backup.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  namespace: openshift-adp
+spec:
+  includedNamespaces:
+    - {{ .Values.backup.includedNamespaces | quote }}
+  snapshotVolumes: {{ .Values.backup.snapshotVolumes }}
+  storageLocation: {{ .Values.backup.storageLocation | quote }}
```
Comment (rasheedamir) [inline application/templates/backup.yaml:11]: @ayeshasiddiqa161 how does user find this?

Comment (ayeshasiddiqa161) [reply application/templates/backup.yaml:11]: The user has to ask the Admin, which storage location to be used for his backup. Right now the DPA is creating dpa-1 location so we are just picking that storage location.

File: application/templates/backup.yaml (hunk 3 above)
Comment (ayeshasiddiqa161) [reply application/templates/backup.yaml:5]: Yes working on it.

File: application/templates/backup.yaml (hunk 4 above)
Comment (rasheedamir) [reply application/templates/backup.yaml:11]: Can't it be picked up dynamically?

File: application/templates/backup.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{{- if (.Values.backup).enabled }}
+apiVersion: velero.io/v1
+kind: Backup
+metadata:
+  name: {{ .Values.backup.backupCRName }}
+  namespace: openshift-adp
```
Comment (aslafy-z) [inline application/templates/backup.yaml:6]: This should be configurable

File: application/templates/backup.yaml (hunk 3 above)
Comment (ayeshasiddiqa161) [reply application/templates/backup.yaml]: Resolved

File: application/templates/backup.yaml (hunk 2 above)
Comment (ayeshasiddiqa161) [reply application/templates/backup.yaml]: Resolved

File: application/templates/backup.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+kind: Backup
+metadata:
+  name: {{ printf "%s-backup" .Values.applicationName | trunc 63 | quote }}
+  namespace: {{ .Values.backup.namespace | quote }}
+spec:
+  includedNamespaces:
+    - {{ printf "%s" .Release.Namespace | quote }}
+  snapshotVolumes: {{ .Values.backup.snapshotVolumes }}
+  storageLocation: {{ .Values.backup.storageLocation | quote }}
+  ttl: {{ .Values.backup.ttl }}
```
Comment (rasheedamir) [inline application/templates/backup.yaml:12]: @ayeshasiddiqa161 we should always set default value for this so, user doesn't have to set always

File: application/templates/backup.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+{{- if (.Values.backup).enabled }}
```
Comment (aslafy-z) [inline application/templates/backup.yaml:1]:
```suggestion
{{- if .Values.backup.enabled }}
```
And add default value file with `backup.enabled: false` and some other sane default values if you want.

Comment (rasheedamir) [reply application/templates/backup.yaml:1]: Agreed @aslafy-z we should do that

**Review by AsfaMumtaz — DISMISSED** [review]

File: application/templates/backup.yaml (hunk 7 above)
Comment (AsfaMumtaz) [reply application/templates/backup.yaml:1]: Added. Resolving comment

File: application/templates/backup.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+apiVersion: velero.io/v1
+kind: Backup
+metadata:
+  name: {{ printf "%s-backup" .Values.applicationName | trunc 63 | quote }}
+  namespace: {{ .Values.backup.namespace | default "openshift-adp" | quote }}
```
[the remark this answers is not in the document]
Comment (AsfaMumtaz) [reply application/templates/backup.yaml]: Removed

**Review by rasheedamir — DISMISSED** [review]
update readme

**Review by AsfaMumtaz — DISMISSED** [review]

**Review by AsfaMumtaz — APPROVED** [review]

**Pull request merged**